### PR TITLE
Create new Command interface and refactor the cli.js entrypoint.

### PR DESCRIPTION
**Pull request by tmeyer2115**
This PR creates a new Command interface that will describe both built-in and
custom Jambo commands. Commands will be kept track of in the CommandRegistry.
Right now, the registry is empty. Later PRs will:
- Provide the plumbing for adding new, custom commands to the registry.
- Make the existing built-in commands implement the Command interface and add
  them to the registry.

This PR also creates a YargsFactory class. This class produces the Yargs CLI
instance that powers Jambo. This class will be simplified greatly once built-in
commands are part of the registry.

Unit tests for the registry and factory will be added later on.

J=SLAP-740
TEST=auto, manual

Ensured all linting and Jest tests still passed. Performed the following tests
manually:

- Verified that 'jambo --help' produced the expected output.
- Verified that 'jambo [command] --help' produced the expected output.
- Verified that command validation worked as expected.
- Ran the majority of the commands to ensure they still worked.

### CODE REVIEW

File: src/models/commands/command.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * An interface that represents a command in the Jambo CLI. 
+ */
+class Command {
+
+    /**
+     * @returns {string} The alias for the command.
+     */
+    getAlias() {}
```
Comment (oshi97) [inline src/models/commands/command.js:9]: What's the difference between alias and name?

File: src/cli.js [hunk 2]
```diff
@@ -23,186 +15,7 @@ try {
...
-   * @param {Object} argv An object containing the command line
-   *                      options and the Jambo defaultTheme.
-   */
-  function addThemeToArgs(argv) {
-    return { ...argv, theme: jamboConfig.defaultTheme };
-  }
+const commandRegistry = new CommandRegistry();
+const yargsFactory = new YargsFactory(commandRegistry);
+const options = yargsFactory.createCLI(jamboConfig);
+options.argv;
```
[the remark this answers is not in the document]
Comment (tmeyer2115) [reply src/cli.js:21]: Every `yargs` example I've seen has this as the last thing appended to the instance. According to the documentation, it gets the args as a plain old object.

File: src/models/commands/command.js (hunk 1 above)
Comment (tmeyer2115) [reply src/models/commands/command.js:9]: Alias seems to be the preferred term when it comes to CLIs (at least in the docs I've read). You execute the command by using its alias on the command line. I'm trying to use domain-specific language here as best I can. Really, alias is just the name used to invoke the command.

File: src/commands/commandregistry.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/**
```
[the remark this answers is not in the document]
Comment (tmeyer2115) [reply src/commands/commandregistry.js:1]: Argh, I don't know why I always have to fight with VSCode on this. Fixed.

File: src/models/commands/argumentmetadata.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    NUMBER: 'number',
+    BOOLEAN: 'boolean'
+}
+Object.freeze(ArgumentType);
+
+/**
+ * A class outlining the metadata for a {@link Command}'s argument. This includes
+ * the type of the argument's values, if it is required, and an optional default.
+ */
+class ArgumentMetadata {
```
[the remark this answers is not in the document]
Comment (tmeyer2115) [reply src/models/commands/argumentmetadata.js:15]: Sure, added it.

**Review by oshi97 — APPROVED** [review]
lgtm!

**Pull request merged**